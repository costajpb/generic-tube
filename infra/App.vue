<script setup lang="ts">
  import { RouterView } from 'vue-router'
  import PageHeader from '@/infra/components/PageHeader.vue';
  import ShowSearch from './components/ShowSearch.vue';
</script>

<template>
  <PageHeader>
    <ShowSearch />
  </PageHeader>
  <RouterView />
</template>
