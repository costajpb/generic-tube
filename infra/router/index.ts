import { createRouter, createWebHistory } from 'vue-router'
import DashboardView from '@/infra/views/DashboardView.vue'

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'dashboard',
      component: DashboardView
    },
    {
      path: '/shows/:id',
      name: 'show-details',
      component: () => import('@/infra/views/ShowDetailsView.vue')
    }
  ]
})

export default router
