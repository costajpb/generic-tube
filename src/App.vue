<script setup lang="ts">
import { RouterView } from 'vue-router'
import PageHeader from '@/src/components/PageHeader.vue'
</script>

<template>
  <PageHeader />
  <RouterView />
</template>

<style>
</style>
