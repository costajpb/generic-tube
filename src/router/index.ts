import { createRouter, createWebHistory } from 'vue-router'
import DashboardView from '@/src/views/DashboardView.vue'

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'dashboard',
      component: DashboardView
    },
    {
      path: '/shows/:id',
      name: 'show-details',
      component: () => import('@/src/views/ShowDetailsView.vue')
    }
  ]
})

export default router
