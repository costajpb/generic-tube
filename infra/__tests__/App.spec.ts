import { flushPromises, mount } from '@vue/test-utils'
import router from '@/infra/router'
import clickOutside from '@/infra/directives/clickOutside'
import App from '@/infra/App.vue'
import snapshowWrapper from './snapshotWrapper'

const mocks = vi.hoisted(async () => ({
  UseCase: await vi.importActual('@/application/shared/use-case'),
  search: vi.fn()
}))

vi.mock('@/application/shared/use-case', async () => {
  const search = (await mocks).search
  const UseCase = (await mocks).UseCase.default as any
  class MockUseCase extends UseCase {
    search(query: string) {
      return search(query)
    }
  }
  return {
    default: MockUseCase
  }
})

vi.mock('@/infra/util/debounce', () => ({
  default: (fn: () => void) => {
    fn()
  }
}))

describe('App', () => {
  it('should render', () => {
    expect(snapshowWrapper(App, { plugins: [router] })).toMatchSnapshot()
  })

  it('should allow searching shows with debounce', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        directives: {
          'click-outside': clickOutside
        }
      }
    })

    await flushPromises()

    const input = wrapper.find('[type=search]')

    const query = 'query'

    ;(input.element as HTMLInputElement).value = query

    input.trigger('input')

    const search = (await mocks).search

    expect(search).toHaveBeenCalledWith(query)
  })
})
