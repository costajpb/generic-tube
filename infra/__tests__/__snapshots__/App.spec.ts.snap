// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="_container_6cc6f3"
  >
    <h1
      class="_title_6cc6f3"
    >
       ABN Amro Tube 
    </h1>
    <div
      class="_extra_6cc6f3"
    >
      
      
      <a
        class="_trigger_22e8e9"
        href="#search"
      >
        Search
      </a>
      <section
        class="_container_122c2d"
        data-is-active="false"
        id="search"
      >
        <div
          class="_actions_122c2d"
        >
          <a
            class="_close_122c2d"
            href="#"
          >
            <span>
              Close
            </span>
          </a>
          <input
            class="_input_122c2d"
            placeholder="Search shows..."
            type="search"
          />
        </div>
        <div
          class="_container_ea0614"
          style="display: none;"
        >
          
          <!--v-if-->
          <ol
            class="_grid_f639ee"
            style="display: none;"
          >
            
            
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            <li
              class="_wrapper_587f33"
            >
              
              
            </li>
            
            
          </ol>
          
        </div>
      </section>
      
      
    </div>
  </header>
  <!---->
  
</div>
`;
